package/MDAnalysis/analysis/base.py

```diff
@@ -389,8 +647,152 @@ def _conclude(self):
+
+        logger.info("Starting preparation")
+        logger.info("Starting analysis loop over %d trajectory frames", self.n_frames)
+
+        if len(indexed_frames) == 0:  # if `frames` were empty in `run` or `stop=0`
+            return self
+
+        frame_indices, frames = (
+            indexed_frames[:, 0],
+            indexed_frames[:, 1],
+        )
+        trajectory = self._trajectory[frames]
+        for idx, ts in enumerate(ProgressBar(trajectory, verbose=verbose, **progressbar_kwargs)):
+            i = frame_indices[idx]
+            self._frame_index = i
+            self._ts = ts
+            self.frames[i] = ts.frame
+            self.times[i] = ts.time
+            self._single_frame()
+        logger.info("Finishing up")
+        return self
+
+    def _setup_computation_groups(
+        self, n_parts: int, start=None, stop=None, step=None, frames=None
+    ) -> list[np.ndarray]:
+        """
+        Splits the trajectory frames, defined by `start/stop/step` or `frames`, into
+        `n_parts` even groups, preserving their indices.
+
+        Parameters
+        ----------
+        n_parts : int
+            number of parts to split the workload into
+        start : int, optional
+            start frame
+        stop : int, optional
+            stop frame
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+        Raises
+        ------
+        ValueError
+            if *both* `frames` and at least one of `start`, `stop`, or `frames`
+            is provided (i.e., set to another value than ``None``)
+
+        Returns
+        -------
+        computation_groups : list[np.ndarray]
+            list of (n, 2) shaped np.ndarrays with frame indices and numbers
+        """
+        if frames is None:
+            start, stop, step = self._trajectory.check_slice_indices(start, stop, step)
+            used_frames = np.arange(start, stop, step)
+        elif not all(opt is None for opt in [start, stop, step]):
+            raise ValueError("start/stop/step cannot be combined with frames")
+        else:
+            used_frames = frames
+
+        if all((isinstance(obj, bool) for obj in used_frames)):
+            arange = np.arange(len(used_frames))
+            used_frames = arange[used_frames]
+
+        # this numpy thing is similar to list(enumerate(frames))
+        enumerated_frames = np.vstack([np.arange(len(used_frames)), used_frames]).T
+
+        return np.array_split(enumerated_frames, n_parts)
+
+    def _configure_client(self, backend: str, n_workers: int, client):
```

Comment:
I decided that private methods better not have default values, since they're only called by a library code where we better be explicit. Hence, there are no default values for them, only type annotations.